### PR TITLE
LPS-198437 Beta tag appears in the wrong position in the object details

### PR DESCRIPTION
**Pull request by Artartur**
Related Issue: [LPS-198437](https://liferay.atlassian.net/browse/LPS-198437)

### CODE REVIEW

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/ObjectDetails.scss [hunk 1]
```diff
@@ -19,10 +19,11 @@
...
 
-.lfr__object-web-edit-object-details-external-data-source-container-beta {
-	left: 200px;
-	margin-top: 7px;
-	position: absolute;
+	&-beta {
+		left: 200px;
+		margin-top: 7px;
+		position: absolute;
+		top: -57px;
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/ObjectDetails.scss:27]: Hey @Artartur, functionally your code is correct and achieving the expected behavior. However, semantically it does not comply with the structure. 

If your goal was to render the beta button next to the panel title, we could have made this change in `EditObjectDetails.tsx` instead use this `top:-57px`.

You should be able to achieve the expected behavior by modifying the way things are done here:

```
					{Liferay.FeatureFlags['LPS-135430'] && (
						<ClayPanel
							collapsable
							defaultExpanded
							displayTitle={
								Liferay.Language.get(
									'external-data-source'
								)}
							displayType="unstyled"
						>
							<ClayPanel.Body>
								<div className="lfr__object-web-edit-object-details-external-data-source-container">
									<ExternalDataSourceContainer
										errors={errors}
										setValues={setValues}
										storageTypes={storageTypes}
										values={values}
									/>

									<div className="lfr__object-web-edit-object-details-external-data-source-container-beta">
										{values.storageType ===
											'salesforce' && <BetaButton />}
									</div>
								</div>
							</ClayPanel.Body>
						</ClayPanel>
					)}
```
[See ](https://clayui.com/docs/components/panel.html)

**Review by igor-franca — CHANGES_REQUESTED** [review]

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/EditObjectDetails.tsx [hunk 2]
```diff
@@ -267,9 +267,20 @@ export default function EditObjectDetails({
...
+										{Liferay.Language.get(
+											'external-data-source'
+										)}
+									</span>
+
+									<div className="lfr__object-web-edit-object-details-external-data-source-panel-container-beta">
+										{values.storageType ===
+											'salesforce' && <BetaButton />}
+									</div>
+								</div>
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/EditObjectDetails.tsx:282]: Hey @Artartur, could you move the conditional up? Thus avoiding an empty div in cases of storageType other than salesforce. i.e:

```
{values.storageType === 'salesforce' && (
	<div className="lfr__object-web-edit-object-details-external-data-source-panel-container-beta">
		<BetaButton />
	</div>
)}
```

**Review by igor-franca — CHANGES_REQUESTED** [review]
hey @Artartur, great progress with this PR! Just a small consideration.

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/ObjectDetails.scss [hunk 3]
```diff
@@ -21,8 +21,12 @@
...
 }
 
-.lfr__object-web-edit-object-details-external-data-source-container-beta {
-	left: 200px;
-	margin-top: 7px;
-	position: absolute;
+.lfr__object-web-edit-object-details-external-data-source-panel {
+	align-items: center;
+	display: flex;
+	flex-direction: row;
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/ObjectDetails.scss:27]: Hi @Artartur, as the default value of flex-direction is row, could you check if this row is really necessary?

**Review by igor-franca — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/ObjectDetails.scss (hunk 3 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/ObjectDetails.scss]: Yeah, you're right. I've changed, thx Igor.

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/EditObjectDetails.tsx (hunk 2 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectDetails/EditObjectDetails.tsx]: Done!

**Review by igor-franca — APPROVED** [review]
Thank you for the work done here @Artartur!

**Comment by igor-franca** [comment]
ci:test:object

**Comment by Artartur** [comment]
ci:test:relevant

**Comment by Artartur** [comment]
ci:test:object

**Comment by igor-franca** [comment]
ci:forward

**Comment by igor-franca** [comment]
ci:stop

**Comment by igor-franca** [comment]
ci:forward:force

**Pull request closed without merging**